internal/lib/testHerd.mli

```diff
@@ -18,15 +18,20 @@
   * would run. *)
 val herd_command: string -> string -> string -> string
 
+type stdout_lines = string list
+type stderr_lines = string list
+
 (** [run_herd herd libdir litmus] runs the binary [herd] with a custom [libdir]
-  * on a [litmus] file, and returns the output with unstable lines removed (e.g.
-  * Time). *)
-val run_herd : string -> string -> string -> string list
+  * on a [litmus] file, and returns the stdout with unstable lines removed (e.g.
+  * Time) and stderr *)
+val run_herd : string -> string -> string -> stdout_lines * stderr_lines
 
-(** [herd_output_matches_expected herd libdir litmus expected] runs the binary
+(** [herd_output_matches_expected herd libdir litmus expected expected_failure] runs the binary
   * [herd] with a custom [libdir] on a [litmus] file, and compares the output
-  * with an [expected] file. *)
-val herd_output_matches_expected : string -> string -> string -> string -> bool
+  * with an [expected] file. if the run writes to stdout then we check [expected_failure].
```

Comment:
Isn't it `If the run writes to stderr` ? (in place of `stdout`).

Reply:
good spot, fixed :)